packages/shorebird_cli/lib/src/commands/account/subscribe_account_command.dart

```diff
@@ -0,0 +1,88 @@
+import 'dart:async';
+
+import 'package:mason_logger/mason_logger.dart';
+import 'package:shorebird_cli/src/command.dart';
+import 'package:shorebird_cli/src/shorebird_config_mixin.dart';
+import 'package:shorebird_code_push_client/shorebird_code_push_client.dart';
+
+/// {@template subscribe_account_command}
+/// `shorebird account subscribe`
+/// {@endtemplate}
+class SubscribeAccountCommand extends ShorebirdCommand
+    with ShorebirdConfigMixin {
+  /// {@macro subscribe_account_command}
+  SubscribeAccountCommand({
+    required super.logger,
+    super.auth,
+    super.buildCodePushClient,
+  });
+
+  @override
+  String get name => 'subscribe';
+
+  @override
+  String get description => 'Sign up for a Shorebird subscription .';
+
+  @override
+  String get summary => '''
+A subscription is required to use Shorebird to create and publish apps.
+Subscriptions are currently \$20 per month and are billed through Stripe.
+Visit ${styleUnderlined.wrap(lightCyan.wrap('https://github.com/shorebirdtech/shorebird/blob/main/TRUSTED_TESTERS.md'))} to learn more.''';
+
+  @override
+  Future<int> run() async {
+    if (!auth.isAuthenticated) {
+      logger
+        ..err('''
+You must be logged in to subscribe.''')
+        ..info('''
+
+If you have a Shorebird account, run ${lightCyan.wrap('shorebird login')} to log in.
+If you don't have a Shorebird account, run ${lightCyan.wrap('shorebird account create')} to create one.''');
+      return ExitCode.software.code;
+    }
+
+    final client = buildCodePushClient(
+      httpClient: auth.client,
+      hostedUri: hostedUri,
+    );
+
+    final User user;
+    try {
+      final maybeUser = await client.getCurrentUser();
```

Comment:
Should we show some sort of progress when performing async operations so that the user has a visual indication that we're processing their request?

Reply:
Done.